src/main.rs

```diff
@@ -296,28 +297,23 @@ fn main() {
         .unwrap();
 
     let conn = Connection::open_with_flags(db_path, OpenFlags::SQLITE_OPEN_READ_WRITE).unwrap();
-    let conn_mtx: Arc<Mutex<Connection>> = Arc::new(Mutex::new(conn));
 
     if !test {
-        conn_mtx
-            .lock()
-            .unwrap()
-            .execute(
-                "CREATE TABLE IF NOT EXISTS tinues (
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS tinues (
         id integer primary key,
         gameid integer NOT NULL REFERENCES games(id),
         size integer,
         plies_to_undo integer,
         tinue_depth integer,
         tinue TEXT)",
-                params![],
-            )
-            .unwrap();
+            params![],
+        )
+        .unwrap();
     }
-
-    // use lambda function here to scope and drop `conn` MutexGuard - otherwhise it will hold the lock forever.
-    let gamerows = (|| {
-        let conn = conn_mtx.lock().unwrap();
+    // Do this step in its own block,
+    // because `stmt` needs to go out of scope before we can take ownership of `conn` again
+    let gamerows = {
```

Comment:
Thanks for making me realize that I didn't need a lambda here! Just something to drop stmt early.